### PR TITLE
[Enhancement][Converter] Add Expression property to MathExpressionConverter and MultiMathExpressionConverter

### LINKED ISSUE #1608: [Enhancement][Converter] Add Expression property to MathExpressionConverter and MultiMathExpressionConverter
## Summary

Add `Expression` property to `MathExpressionConverter` and `MultiMathExpressionConverter`. I love this `Converter`, but based on the way that I normally use it,  it would be great to be able to set up the `Expression` value in the Resouce declaration.

## API Changes

Added:
- `public string Expression { get; set; } = string.Empty;`

## Intended Use Case

```
<xct:MathExpressionConverter x:Key="DivideByTwoConverter" Expression="x/2"  />
```

## Who Will Do The Work?

- [x] I am willing to take this on myself
- [ ] Just putting this out there for someone to pick up

### PR DESCRIPTION
**Pull request by GUOLDEV**
### Bugs Fixed ###

- Fixes #1608 

### API Changes ###

Added:
- `public string Expression { get; set; }`

<!-- List all API changes here (or just put None), example:

Added: 
 
- `string Class.Property { get; set; }`
- `void Class.Method();`

Changed:

 - `object Cell.OldPropertyName` => `object Cell.NewPropertyName`
 
If there is an entirely new API, then you can use a more verbose style:

```csharp
public static class NewClass {
    public static int SomeProperty { get; set; }
    public static void SomeMethod(string value);
}
``` -->

### Behavioral Changes ###

<!-- Describe any non-bug related behavioral changes that may change how users app behaves when upgrading to this version of the codebase. -->

### PR Checklist ###
<!-- Please check all the things you did here and double-check that you got it all, or state why you didn't do something -->
- [x] Has a linked Issue, and the Issue has been `approved`
- [x] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description)
- [x] Rebased on top of main at time of PR
- [x] Changes adhere to coding standard
<!-- If at all possible, please update/add the documentation on the repo below. We would very much appreciate that. If you are unable to, please consider at least opening an issue on the repo below so we know that Docs still need to be adjusted/created. Thanks! <3 -->
- [x] Updated [documentation](https://github.com/MicrosoftDocs/xamarin-communitytoolkit)

### CODE REVIEW

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Converters/MathExpressionConverter/MathExpressionConverter.shared.cs [hunk 1]
```diff
@@ -7,6 +7,11 @@ namespace Xamarin.CommunityToolkit.Converters
 {
 	public class MathExpressionConverter : ValueConverterExtension, IValueConverter
 	{
+		/// <summary>
+		/// The expression to calculate.
+		/// </summary>
+		public string Expression { get; set; } = string.Empty;
```
Comment (AndreiMisiukevich) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Converters/MathExpressionConverter/MathExpressionConverter.shared.cs:13]:
```suggestion
		public string? Expression { get; set; }
```

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Converters/MathExpressionConverter/MathExpressionConverter.shared.cs [hunk 2]
```diff
@@ -17,7 +22,7 @@ public class MathExpressionConverter : ValueConverterExtension, IValueConverter
 		/// <returns>A <see cref="double"/> The result of calculating an expression.</returns>
 		public object? Convert(object? value, Type? targetType, object? parameter, CultureInfo culture)
 		{
-			if (parameter is not string expression)
+			if ((parameter ?? Expression ?? string.Empty) is not string expression)
```
Comment (AndreiMisiukevich) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Converters/MathExpressionConverter/MathExpressionConverter.shared.cs:25]:
```suggestion
			if ((parameter ?? Expression) is not string expression)
```

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Converters/MathExpressionConverter/MultiMathExpressionConverter.shared.cs [hunk 3]
```diff
@@ -8,6 +8,11 @@ namespace Xamarin.CommunityToolkit.Converters
 {
 	public class MultiMathExpressionConverter : MultiValueConverterExtension, IMultiValueConverter
 	{
+		/// <summary>
+		/// The expression to calculate.
+		/// </summary>
+		public string Expression { get; set; } = string.Empty;
```
Comment (AndreiMisiukevich) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Converters/MathExpressionConverter/MultiMathExpressionConverter.shared.cs:14]:
```suggestion
		public string? Expression { get; set; }
```

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Converters/MathExpressionConverter/MultiMathExpressionConverter.shared.cs [hunk 4]
```diff
@@ -18,7 +23,7 @@ public class MultiMathExpressionConverter : MultiValueConverterExtension, IMulti
 		/// <returns>A <see cref="double"/> The result of calculating an expression.</returns>
 		public object? Convert(object[]? values, Type? targetType, object? parameter, CultureInfo culture)
 		{
-			if (parameter is not string expression)
+			if ((parameter ?? Expression ?? string.Empty) is not string expression)
```
Comment (AndreiMisiukevich) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Converters/MathExpressionConverter/MultiMathExpressionConverter.shared.cs:26]:
```suggestion
			if ((parameter ?? Expression) is not string expression)
```

Comment (GUOLDEV) [reply src/CommunityToolkit/Xamarin.CommunityToolkit/Converters/MathExpressionConverter/MultiMathExpressionConverter.shared.cs]: Done!

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Converters/MathExpressionConverter/MultiMathExpressionConverter.shared.cs (hunk 3 above)
Comment (GUOLDEV) [reply src/CommunityToolkit/Xamarin.CommunityToolkit/Converters/MathExpressionConverter/MultiMathExpressionConverter.shared.cs]: Done!

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Converters/MathExpressionConverter/MathExpressionConverter.shared.cs (hunk 2 above)
Comment (GUOLDEV) [reply src/CommunityToolkit/Xamarin.CommunityToolkit/Converters/MathExpressionConverter/MathExpressionConverter.shared.cs]: Done!

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Converters/MathExpressionConverter/MathExpressionConverter.shared.cs (hunk 1 above)
Comment (GUOLDEV) [reply src/CommunityToolkit/Xamarin.CommunityToolkit/Converters/MathExpressionConverter/MathExpressionConverter.shared.cs]: Done!

**Comment by GUOLDEV** [comment]
@AndreiMisiukevich Thanks for checking the PR and suggestions 👍   all done and I have also updated the documentation [PR](https://github.com/MicrosoftDocs/xamarin-communitytoolkit/pull/137)

**Review by AndreiMisiukevich — APPROVED** [review]

**Comment by GUOLDEV** [comment]
Hey @AndreiMisiukevich, is there anything else that I can do to get this PR merged?

**Comment by GUOLDEV** [comment]
@jfversluis @pictos @brminnick I know you guys are really busy 😃 , but could you review this PR when you have a chance? Thanks in advance 🙏

**Pull request merged**